### PR TITLE
Local testing parity

### PR DESCRIPTION
**Pull request by spectorar**
These changes are in addition to the ones currently already in develop, represented by this PR: https://github.com/cloudmine/node-coderunner/pull/4

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -23,7 +23,7 @@ test:
 	$(MAKE) integration
 	$(MAKE) travis-cov
 	$(MAKE) lint
-	$(MAKE) check-dependencies
+	$(MAKE) check-dependencies || true
```
Comment (Splat) [inline Makefile]: Can we add a comment in here for justification

File: lib/server.coffee [hunk 2]
```diff
@@ -43,26 +44,28 @@ class Server
   ###
   _setupRoutes: ->
     @_names()
-    if isTruthy process.env['LOCAL_TESTING']
-      @_setupLocalTestingRoutes()
-    else
+    if isTruthy process.env['CLOUDMINE']
```
Comment (Splat) [inline lib/server.coffee:49]: This basically exposes to the client the environment variable. Is there a mechanism to make this non-mutable? I feel like this should come from a config environment. Kinda scares me that we make this visible to the implementer.

Comment (Splat) [inline lib/server.coffee:49]: Brian and Arthur walked me off the ledge. This should be fine. However I still would like a comment that explains this and the possible states of this environment variable.

File: lib/server.coffee [hunk 3]
```diff
@@ -76,20 +79,25 @@ class Server
...
-  _setupPutAndPostRoute: (path, handler) ->
+  _setupPutAndPostRoute: (path, handler, timeout) ->
     @server.route
       method: ['PUT', 'POST']
       path: path
       config:
         payload:
           maxBytes: MAX_PAYLOAD_BYTES
+        timeout:
+          server: timeout or null
```
Comment (Splat) [inline lib/server.coffee:95]: can we get a comment justifying the null case. Probably just an overall comment for the entire method.

File: lib/server.coffee [hunk 4]
```diff
@@ -76,20 +79,25 @@ class Server
...
 
-  _setupGetRoute: (path, handler) ->
+  _setupGetRoute: (path, handler, timeout) ->
     @server.route
       method: ['GET']
       path: path
       handler: handler
+      config:
+        timeout:
+          server: timeout or null
```
Comment (Splat) [inline lib/server.coffee:107]: can we get a comment justifying the null case. Probably just an overall comment for the entire method.

File: test/integration/server.coffee [hunk 5]
```diff
@@ -31,6 +31,7 @@ describe 'Server', ->
 
   describe 'Deployed API', ->
     before (done)->
+      process.env['CLOUDMINE'] = 1
```
Comment (Splat) [inline test/integration/server.coffee:34]: This is setting to a number where prior checks look for truthy? I'm guessing 1 is truthy?

Comment (spectorar) [inline test/integration/server.coffee:34]: correct. and 0/1 is what is used in real life so I kept that going.

**Comment by Splat** [comment]
I'm good to go on this!

**Comment by Splat** [comment]
I am good to go on this. Once we fix the build fail that is

**Comment by spectorar** [comment]
Build fixed!

**Pull request merged**